### ISSUE TITLE
Could PackageCompiler install a compiler when run on Linux?

### ISSUE THREAD

**Issue by PGS62** [JuliaLang/PackageCompiler.jl#647]
When running PackageCompiler under Windows, "A suitable compiler will be automatically installed the first time it is needed". However, that's not the case on Linux or MacOS. So my suggestion is to make PackageCompiler do the automatic installation of a compiler if that's necessary when running on Linux or MacOS.

For my current project, which includes automated creation of a system image, that would make things simpler.

**Comment by KristofferC** [comment]
It could download one on demand if it doesn't find a local one.

**Comment by DilumAluthge** [comment]
@staticfloat On non-Windows systems, could we lazily download `Clang_jll`?

**Comment by staticfloat** [comment]
Yes, but the complexity is that we don't just need a compiler, we also need the supporting libc and header files and whatnot.  While we have those collected for BinaryBuilder already, it's not in a very modular, easy-to-get format.  It's something I've been wanting to work on for a while, but realistically is many months away from being at the top of my TODO list.
